### DISCUSSION TITLE
Re-use fetched variables in a where query

### DISCUSSION THREAD

**Discussion by notscar** [prisma/prisma#15171] [Q&A] [answered]
Is it possible to specify a where request that uses a variable fetched in the same query?

for example:

```prisma
model One {
    id String @id @unique
    somethingElse Int
    categoryId Int
    category Category ? @relation(fields: [categoryId], references: [id], onDelete: Cascade, onUpdate: Cascade)
}

model Category {
    id String @id @unique @default(uuid())
    something Int
}
```

```js
await prisma.one.findUnique({
    where: {
        id: 2
    },
    select: {
        category: {
            where: {
                something: somethingElse
            }
        }
    }
})
```

**Comment by austincrim** [comment, marked as answer]
Hey @NotScar 👋,

This isn't currently possible without [`$queryRaw`](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw), but we recently added the [`fieldReference`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#compare-columns-in-the-same-table) preview feature, which is the first step on making this pattern work with regular Prisma queries.

It would be great if you could leave a comment on [this issue](https://github.com/prisma/prisma/issues/15068) describing your use case with a little more detail. These examples are really helpful when designing new features.

Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
